python_modules/dagster/dagster/_core/definitions/asset_selection.py

```diff
@@ -303,13 +305,14 @@ def resolve(
             asset_graph = AssetGraph.from_assets(all_assets)
 
         resolved = self.resolve_inner(asset_graph)
-        resolved_source_assets = asset_graph.source_asset_keys & resolved
-        resolved_regular_assets = resolved - asset_graph.source_asset_keys
-        check.invariant(
-            not (len(resolved_source_assets) > 0 and len(resolved_regular_assets) > 0),
-            "Asset selection specified both regular assets and source assets. This is not"
-            " currently supported. Selections must be all regular assets or all source assets.",
-        )
+        if not allow_regular_and_source_assets:
```

Comment:
It probably makes sense to post this part as a separate change

Reply:
for reasons of different review audience

Reply:
roger

Reply:
Moved that to https://github.com/dagster-io/dagster/pull/18681